cnf-certification-test/performance/suite.go

```diff
@@ -79,6 +89,32 @@ var _ = ginkgo.Describe(common.PerformanceTestKey, func() {
 	// Scheduling related tests ends here
 })
 
+func testLimitedUseOfExecProbes(env *provider.TestEnvironment) {
+	var compliantObjects []*testhelper.ReportObject
+	var nonCompliantObjects []*testhelper.ReportObject
+	for _, put := range env.Pods {
+		counter := 0
+		for _, cut := range put.Containers {
+			if cut.HasExecProbes() {
+				tnf.ClaimFilePrintf("Pod %s/Container %s use of exec probes, InitialDelaySeconds of LivenessProbe: %s,  InitialDelaySeconds of ReadinessProbe: %s, InitialDelaySeconds of StartupProbe: %s", cut.Podname, cut,
+					cut.LivenessProbe.InitialDelaySeconds, cut.ReadinessProbe.InitialDelaySeconds, cut.StartupProbe.InitialDelaySeconds)
+				counter++
```

Comment:
If the container has `Exec` probes, check each `Liveness`, `Readiness`, and `Startup` probes for their PeriodSeconds <= 10.  If their `Exec` probe has a PeriodSeconds > 10, it's a fail.

Also, we need to count the number of `Exec` probes in the entire cluster, so just move the counter outside of the Pods loop.  There cannot be more than 10 `Exec` probes in the entire cluster.